hack/update-codegen.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash -e
+
+set -x
+
+# set the passed in directory as a usable GOPATH
+# that deepcopy-gen can operate in
+ensure-temp-gopath() {
+	fake_gopath=$1
+
+	# set up symlink pointing to our repo root
+	fake_repopath=$fake_gopath/src/github.com/openshift/cloud-credential-operator
+	mkdir -p "$(dirname "${fake_repopath}")"
+	ln -s "$REPO_FULL_PATH" "${fake_repopath}"
+}
+
+SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
+REPO_FULL_PATH=$(realpath ${SCRIPT_ROOT})
+cd ${REPO_FULL_PATH}
+
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../../../k8s.io/code-generator)}
+
+verify="${VERIFY:-}"
+
+valid_gopath=$(realpath $REPO_FULL_PATH/../../../..)
+if [[ "$(realpath ${valid_gopath}/src/github.com/openshift/cloud-credential-operator)" == "${REPO_FULL_PATH}" ]]; then
```

Comment:
It was an optimization suggested by @staebler so that we wouldn't have to wait for the various pieces to be downloaded to the GOPATH if your usable GOPATH already had them in there. Works out that the TMP_DIR holds around 174MB of data on disk.

I'll un-WIP and squash some things up and reduce the verbosity of the deepcopy generation in a moment.